pulp_ansible/app/downloaders.py

```diff
@@ -51,25 +52,31 @@ def __init__(
         self.silence_errors_for_response_status_codes = silence_errors_for_response_status_codes
         super().__init__(url, **kwargs)
 
-    def raise_for_status(self, response):
+    def raise_for_status(self, response, handle_401=False):
         """
         Raise error if aiohttp response status is >= 400 and not silenced.
 
+        Args:
+            handle_401(bool): If true, catch 401, request a new token and retry.
+
         Raises:
             FileNotFoundError: If aiohttp response status is 404 and silenced.
             aiohttp.ClientResponseError: If the response status is 400 or higher and not silenced.
 
         """
         silenced = response.status in self.silence_errors_for_response_status_codes
 
-        if not silenced:
+        if handle_401 and response.status == 401:
+            self.registry_auth["bearer"] = None
```

Comment:
This shouldn't really be set here. Due to separation of concerns this should handle the silencing but not what happens if it's silenced. I gotta look closer to suggest what to do instead.